### PR TITLE
Add documentation for all current parameters

### LINKED ISSUE #27: Document all the available config options
New options were added and need to be documented:

- [x] #4 / #22
- [x] #9 / #20
- [x] #3 / #29 
- [x] #26 / #30 
- [x] #32 / #33

### PR DESCRIPTION
**Pull request by dominykas**
Closes #27.

This also documents some params which are [at the time of opening this] still in review, so it should not be merged before these:

- [x] https://github.com/pkgjs/action/pull/33
- [x] https://github.com/pkgjs/action/pull/29

### CODE REVIEW

File: docs/USAGE.md [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+
+## Secrets (allowed [via `jobs.<job_id>.secrets`](https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#jobsjob_idsecrets))
+
+### `test-secrets`
+
+A JSON object with the secret environment variables to be set for the 'Run tests' step. Values will be masked in output.
+
+Note: make sure that your secrets are valid JSON strings (you can escape them by using [`toJSON` expression](https://docs.github.com/en/actions/learn-github-actions/expressions#tojson))
+
+⚠️ Important: ⚠️ while care has been taken to mask the secret values in log output (the input JSON as a whole and the individual values), there might exist a chance of leaking them via the publicly available UIs, as there is no official Github Actions support for passing arbitrary environment variables and secrets into re-usable workflows. The implementation of `test-secrets` is, generally, a workaround for the limitation that re-usable workflows do not receive the environment of the calling workflow.
```
Comment (Eomm) [inline docs/USAGE.md:254]:
```suggestion
⚠️ Security Warning: ⚠️

While care has been taken to mask the secret values in log output (the input JSON as a whole and the individual values), there might exist a chance of leaking them via the publicly available UIs. 
There is no official Github Actions support for passing arbitrary environment variables and secrets into re-usable workflows. 
The implementation of `test-secrets` is, generally, a workaround for the limitation that re-usable workflows do not receive the environment of the calling workflow.
```

File: docs/USAGE.md [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+    with:
+      runs-on: |
+        - ubuntu-latest
+        - macos-latest
+```
+
+
+### `strategy-fail-fast`
+
+Set to `true` to cancel the workflow as soon as the first matrix combo fails. Note: ⚠️ default here is different than the Github Actions default.
```
Comment (Eomm) [inline docs/USAGE.md:192]:
```suggestion
Set to `true` to cancel the workflow as soon as the first matrix combo fails.

Note: ⚠️ default here is different than the Github Actions default.
```

File: docs/USAGE.md [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+- [`include`](#include)
+- [`runs-on`](#runs-on)
+
+
+#### Example
+
+```yaml
+jobs:
+  test:
+    uses: pkgjs/action/.github/workflows/node-test.yaml@main
```
Comment (Eomm) [inline docs/USAGE.md:37]: I think that documenting the `node-test.yaml@main` could hurt us when there will be some breaking changes when this GHA will evolve.

Do you think that cutting the `v1` tag would be better?

Adding this workflow:

https://github.com/fastify/github-action-merge-dependabot/blob/main/.github/workflows/release.yml#L25

to this repository would make it even easier to release

Comment (dominykas) [reply docs/USAGE.md:37]: Yes, I will tag `v1` as soon as everything is ready.

I haven't looked into the details of that workflow, but I've seen it exists - will definitely consider it when the time comes.

**Review by Eomm — APPROVED** [review]

**Pull request merged**